File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int si = 0;
        int ei = nums.length-1;
        int idx = ei+1;
        while(si < ei){
            int mid = si + (ei - si)/2;
            
            // Conditions:
            if(nums[mid] == target){
                return mid;
            }
            else if(nums[mid] > target){
                idx = mid;
                ei = mid -1;
            }
            else{
                si = mid +1;
            }
        }
        return (si==ei && nums[si] >= target)? si : idx;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    # Binary Search:
    def searchInsert(self, nums: List[int], target: int) -> int:
        si = 0
        ei = len(nums) - 1
        pans = len(nums)
        while si <= ei:
            mid = math.floor(si + (ei - si) / 2)
            if nums[mid] == target: return mid
            elif nums[mid] > target:
                pans = mid
                ei = mid -1
            else: si = mid +1

        return pans